python/graphstorm/dataloading/sampler.py

```diff
@@ -389,3 +392,247 @@ def sample_blocks(self, g, seed_nodes, exclude_eids=None):
             blocks.insert(0, block)
 
         return seed_nodes, output_nodes, blocks
+
+class _FileSampler:
+    r"""File Sampler Interface.
+
+    Parameters:
+    ---------
+    dataset_path : str
+        Path to the data files.
+        Mutually exclusive with :attr:`files`.
+    """
+
+    def __init__(self, dataset_path=None):
+        self.dataset_path = None
+        self.files = None
+
+        if not dataset_path:
+            raise ValueError(
+                "Dataset_path={}, should be specified.".format(
+                    dataset_path
+                )
+            )
+        else:
+            if not isinstance(dataset_path, str):
+                raise TypeError(
+                    "dataset_path should be a str, but got " "dataset_path={}".format(dataset_path)
+                )
+
+            self.dataset_path = dataset_path
+            self.files = [
+                os.path.join(dataset_path, f)
+                for f in os.listdir(dataset_path)
+                if os.path.isfile(os.path.join(dataset_path, f))
+            ]
+            self.files = [f for f in self.files if os.path.getsize(f)>0]
+            self.files.sort()
+
+            if len(self.files)==0:
+                raise ValueError (
+                    "no non-empty files found at top directory {}.".format(dataset_path))
+
+        self.num_files = len(self.files)
+        print ("found {} files from {}".format(self.num_files, dataset_path))
+
+    def __len__(self):
+        return self.num_files
+
+    def __iter__(self):
+        raise NotImplementedError
+
+    def __next__(self):
```

Comment:
can you explain what this function returns?

Reply:
This function returns the index of the next data file.

Reply:
Is this an abstract class?
Can you add comments specifying what is expected as the return value?

Reply:
yes it's abstract class. I'll add comments.